rpc/rpcserver/server.go

```diff
@@ -1398,31 +1398,29 @@ func (t *ticketbuyerServer) StartAutoBuyer(ctx context.Context, req *pb.StartAut
 	}
 	params := wallet.ChainParams()
 
-	var votingAddress string
+	var votingAddress dcrutil.Address
 	if req.VotingAddress != "" {
-		votingAddr, err := decodeAddress(req.VotingAddress, params)
+		votingAddress, err = decodeAddress(req.VotingAddress, params)
 		if err != nil {
 			return nil, err
 		}
-		votingAddress = votingAddr.String()
 	}
 
-	var poolAddress string
+	var poolAddress dcrutil.Address
 	if req.PoolAddress != "" {
-		poolAddr, err := decodeAddress(req.PoolAddress, params)
+		poolAddress, err = decodeAddress(req.PoolAddress, params)
 		if err != nil {
 			return nil, err
 		}
-		poolAddress = poolAddr.String()
 	}
 
 	poolFees := req.PoolFees
 	switch {
-	case poolFees == 0 && poolAddress != "":
+	case poolFees == 0 && poolAddress.String() != "":
```

Comment:
This and the two checks below will panic with nil addresses.  It must compare the address against nil, not the address string (which can't be created without the address).